### ISSUE TITLE
Auto-speed is broken again 2.3.57.1

### ISSUE THREAD

**Issue by NameOfTheDragon** [supermerill/SuperSlicer#1629]
### Version
2.3.57.1

### Operating system type + version
Windows 10 Pro

### Behavior
Slice an object using a profile with auto-speed enabled.
![image](https://user-images.githubusercontent.com/1268490/136707784-d230fb1b-0309-418a-b04b-25a31f6349c4.png)

Also note filament overrides:
![image](https://user-images.githubusercontent.com/1268490/136707810-347a8265-e686-4e60-abc7-da71de15347d.png)

Switch to GCODE view, and examine the print speeds.
![image](https://user-images.githubusercontent.com/1268490/136707986-6339d112-93f6-47f7-b24c-79e82942d3bf.png)

The generated gcode has the entire model printing at 0.7 mm/s (which is way below the configured printer minimum print speed of 15 mm/s) - it should several orders of magnitude faster, especially the infill.
![image](https://user-images.githubusercontent.com/1268490/136707906-733dede5-50a0-4bec-a021-efe6bbcfd9fb.png)

#### Project File (.3MF) where problem occurs
Use Calibration -> Calibration Cube -> Standard Cube, and set speeds as in the screen shots above.

**Comment by neophyl** [comment]
The reason a project file is requested is that there are potentially other areas that can effect things. For example filament settings can over ride other speed settings. That’s why ALL the profile settings are needed.

**Comment by supermerill** [comment]
You set 0 to gap_fill speed, bad idea as there are many very small extrusion (see tooltip).

**Comment by jankolkmeier** [comment]
I noticed something (that I thought was odd) about autospeed for gap moves and had this image prepared for a new bug report, but I just found this discussion which seems to be talking about the same issue:

![Gap Autospeed affects other feature speed](https://i.imgur.com/eZcjCp1.png)

While this may be working as intended, I have to agree with others here that it is unintuitive.
My primary confusion is this: After setting gap speed to auto, I notice that print time has doubled. I want to inspect what's going on using the slicer preview. What I expect to see is that the percentage/time of "gap" extrusions has significantly increased (the white extrusions in the preview). Instead, I actually see that gap move time has decreased, but times of other moves, such as infill and perimenter, have significantly increased - although I made no change regarding those speeds.

The suggested change to the tooltip unfortunately does not clear up the confusion for me. In fact, it confuses me more: Why would thin gaps negatively affect the speed when using autospeed? Thin extrusions require less flow and thus should allow for faster moves? Apologies if I am missing or confusing some context required for understand this, @supermerill .

As a side note, the main reason I was interested in autospeed for gaps was to limit speed on extremely fat gaps, as I had previously exceeded my extruders capabilities with a fixed gap speed on those fat gaps.

**Comment by supermerill** [comment]
Thin extrusion hit the max speed with very little flow rate.
The whole point of the auto-speed is to keep the same flow for the whole print if possible.
So it keep this extremely slow flow rate for everything.

**Comment by jankolkmeier** [comment]
Thanks a lot for clearing it up, I didn't realize the purpose of autospeed is to prioritize *"constant pressure"* over *"as close to the max volumetric speed as possible"*. I was simply misusing this feature then. Reading it all back now makes sense, and I should have more closely read the tooltips under the  "Auto Speed (advanced)" section. 

I've tried to come up with a wording for a warning or popup that explains the issue without leaving room for confusion about the Auto Speed feature, but I'm afraid it is too wordy:


> **Auto Speed** will try to maintain a constant flow rate accross all print moves.
It is not recommended to include gap moves to the **Auto Speed** calculation (by setting this value to 0).
Very thin gap extrusions will often not max out the flow rate of your printer.
As a result, this will cause **Auto Speed** to lower the speeds of all other print moves to match the low flow rate of these thin gaps.

PS: To solve my original issue about fat gap moves specifically exceeding maximum flow, instead of Auto Speed I found that I can limiting maximum flow through filament_max_volumetric_speed.
